gax/src/main/java/com/google/api/gax/rpc/BidiStreamingCallable.java

```diff
@@ -44,22 +44,85 @@
 
   protected BidiStreamingCallable() {}
 
+  public abstract ClientStream<RequestT> call(
+      ResponseObserver<ResponseT> responseObserver,
+      ClientStreamCallBack<RequestT> onReady,
+      ApiCallContext context);
+
+  public void call(
+      final BidiStreamObserver<RequestT, ResponseT> bidiObserver, ApiCallContext context) {
+    call(
+        bidiObserver,
+        new ClientStreamCallBack<RequestT>() {
+          @Override
+          public void call(ClientStream<RequestT> stream) {
+            bidiObserver.onReady(stream);
+          }
+        },
+        context);
+  }
+
+  public BidiStream<RequestT, ResponseT> call(ApiCallContext context) {
+    BidiStream<RequestT, ResponseT> stream = new BidiStream<>();
+    ClientStream<RequestT> clientStream = call(stream.observer(), context);
+    stream.setClientStream(clientStream);
+    return stream;
+  }
+
+  public ClientStream<RequestT> call(
+      ResponseObserver<ResponseT> responseObserver, ApiCallContext context) {
+    return call(
+        responseObserver,
+        new ClientStreamCallBack<RequestT>() {
+          @Override
+          public void call(ClientStream<RequestT> stream) {
+            // no op
+          }
+        },
+        context);
+  }
+
+  public ClientStream<RequestT> call(ResponseObserver<ResponseT> responseObserver) {
+    return call(responseObserver, null);
+  }
+
   /**
    * Conduct a bidirectional streaming call with the given {@link ApiCallContext}.
    *
    * @param responseObserver {@link ApiStreamObserver} to observe the streaming responses
    * @param context {@link ApiCallContext} to provide context information for the RPC call.
    * @return {@link ApiStreamObserver} which is used for making streaming requests.
    */
-  public abstract ApiStreamObserver<RequestT> bidiStreamingCall(
-      ApiStreamObserver<ResponseT> responseObserver, ApiCallContext context);
+  @Deprecated
+  public ApiStreamObserver<RequestT> bidiStreamingCall(
+      ApiStreamObserver<ResponseT> responseObserver, ApiCallContext context) {
+    final ClientStream<RequestT> stream =
+        call(new ServerStreamingCallable.ApiStreamObserverAdapter<>(responseObserver), context);
```

Comment:
Another case of an internal class being used in a separate context...

Reply:
Done